### PR TITLE
D2IQ-71117: adds ColorCodedBadge component

### PR DESCRIPTION
**Pull request by mperrotti**
Closes D2IQ-71117

## Testing

Just look at Storybook

## Screenshots

Screen Shot 2020-08-27 at 2 18 07 PM

## Checklist for PR creator

- [x] If any new components were added, there are exported from `packages/index.ts`
- [x] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [x] Info for applicable sections above is provided

### CODE REVIEW

File: packages/badge/components/ColorCodedBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          className={tintContent(color)}
+          width={parseInt(ICON_SIZE, 10)}
+          height={parseInt(ICON_SIZE, 10)}
+          viewBox={`0 0 ${parseInt(ICON_SIZE, 10)} ${parseInt(ICON_SIZE, 10)}`}
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <circle
+            cx={parseInt(ICON_SIZE, 10) / 2}
+            cy={parseInt(ICON_SIZE, 10) / 2}
+            r={parseInt(ICON_SIZE, 10) / 2}
```
Comment (pierrebeitz) [inline packages/badge/components/ColorCodedBadge.tsx:36]: just a nitpick that most likely will never matter: 
if we'd pull out `parseInt(ICON_SIZE, 10)` into a constant, we would not recalculate those things a couple times when instantiating a badge.

same would go fo the `cirle`: it's a constant, so we could pull it out of the actual badge. the whole application would then refer to the same `circle`.

File: packages/badge/stories/badge.stories.tsx [hunk 2]
```diff
@@ -47,4 +80,25 @@ storiesOf("Graphic elements|Badge", module)
         text: "Button badges for when we want add click event"
       }
     }
-  );
+  )
+  .add("ColorCodedBadge", () => {
+    const color = select("Color", colors, blue);
```
Comment (pierrebeitz) [inline packages/badge/stories/badge.stories.tsx:85]: neat! 👍

**Review by pierrebeitz — APPROVED** [review]
just had a brief look. not a thorough approval.

**Comment by natmegs** [comment]
Argh sorry I was a bit premature on the merge - only noticed after that Pierre said his was not a thorough approval. Let me know if I should undo this!

**Pull request merged**